src/CarbonAware.Aggregators/src/CarbonAware/CarbonAwareAggregator.cs

```diff
@@ -32,8 +32,17 @@ public async Task<IEnumerable<EmissionsData>> GetEmissionsDataAsync(IDictionary
         {
             DateTimeOffset end = GetOffsetOrDefault(props, CarbonAwareConstants.End, DateTimeOffset.Now.ToUniversalTime());
             DateTimeOffset start = GetOffsetOrDefault(props, CarbonAwareConstants.Start, end.AddDays(-7));
+            bool best = GetBestOrFalse(props, CarbonAwareConstants.Best);
             _logger.LogInformation("Aggregator getting carbon intensity from data source");
-            return await this._dataSource.GetCarbonIntensityAsync(GetLocationOrThrow(props), start, end);
+
+            var result = await this._dataSource.GetCarbonIntensityAsync(GetLocationOrThrow(props), start, end);
+
+            if (best)
+            {
+                var bestResult = GetOptimalEmissions(result);
+                return new EmissionsData[] { bestResult };
```

Comment:
Given that `best` is conceptually a single item, this feels like it should have its own method `public async Task<EmissionsData> GetBestEmissionsDataAsync(IDictionary props)` and be wired up through the controller to reflect that.

Having this conditional path that also has to shoehorn the typing signals that maybe this method is trying to do too much.

Reply:
I changed it around already, so I would consider this resolved.